### ISSUE TITLE
dnf does not upgrade to dedicated package version on rhel8 (except allow_downgrade: yes is set)

### ISSUE THREAD

**Issue by slm0n87** [ansible/ansible#74209]
### Summary

I want to update an rpm to a dedicated version using the buildin dnf task.  
The version is specified in the "name:" field, but the rpm stays at the old version and ansible does not upgrade the package to the defined version.

### Issue Type

Bug Report

### Component Name

dnf

### Ansible Version

```console
Tested with two version:

Ansible 2.9.14 
and:
$ ansible --version
ansible 2.10.7
  config file = /Users/xxxxx/git/ansible/ansible.cfg
  configured module search path = ['/Users/xxxxx/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/3.2.0/libexec/lib/python3.9/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.9.2 (default, Mar 26 2021, 23:27:12) [Clang 12.0.0 (clang-1200.0.32.29)]
```

### Configuration

```console
$ ansible-config dump --only-changed
ANSIBLE_PIPELINING(/Users/xxxxx/git/ansible/ansible.cfg) = True
DEFAULT_CALLBACK_WHITELIST(/Users/xxxxx/git/ansible/ansible.cfg) = ['profile_tasks']
DEFAULT_FORKS(/Users/xxxxx/git/ansible/ansible.cfg) = 20
DEFAULT_HOST_LIST(/Users/xxxxx/git/ansible/ansible.cfg) = ['/Users/xxxxx/git/ansible/hosts']
DEFAULT_LOG_PATH(/Users/xxxxx/git/ansible/ansible.cfg) = /var/log/ansible/ansible.log
DEFAULT_MANAGED_STR(/Users/xxxxx/git/ansible/ansible.cfg) = Ansible managed
DEFAULT_TRANSPORT(/Users/xxxxx/git/ansible/ansible.cfg) = smart
HOST_KEY_CHECKING(/Users/xxxxx/git/ansible/ansible.cfg) = False
PERSISTENT_CONNECT_TIMEOUT(/Users/xxxxx/git/ansible/ansible.cfg) = 30
```

### OS / Environment

RHEL 8

### Steps to Reproduce

```yaml (paste below)
# variables rpm_name and rpm_version needs to be defined
- name: Rollout rpm to a dedicated version
  dnf:
    name: "{{ rpm_name }}-{{ rpm_version}}.el{{ ansible_distribution_major_version }}"
    update_cache: yes
```
1. Install an older version of an package
2. Install a newer version of an package

### Expected Results

Ansible should update the rpm to the version specified at step 2.

### Actual Results

```console
Ansible Output after Step 2 is:
ok: [myrhel8host]
```

### Code of Conduct

I agree to follow the Ansible Code of Conduct

**Comment by Slm0n87** [comment]
I found a workaround: If `allow_downgrade: yes` is defined for the dnf task then the package gets upgraded.  
Of course this is still an unexpected behaviour.

**Comment by samdoran** [comment]
Can you provide the version of the installed packaged and the version trying to be installed?

needs_info

**Comment by ansibot** [comment]
@Slm0n87 This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@Slm0n87 You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
